File: java/program/Clothing.java
// kelas Clothing memiliki inheritance dari product
public class Clothing extends Product{
    

    private String size;
    private String material;
    private String gender;

    // empty constructor
    public Clothing() {

    }

    // constructor dengan set attribute
    public Clothing(String size, String material, String gender) {
        this.size = size;
        this.material = material;
        this.gender = gender;
    }
    
    // constructor dengan set attribute kelas clothing dan attribute kelas parent (Product)
    public Clothing(String idProduct, String name, String brand, int price, String size, String material,
            String gender) {
        super(idProduct, name, brand, price);   // memanggil constructor superclass/parent
        this.size = size;
        this.material = material;
        this.gender = gender;
    }
    // getters & setters
    public String getSize() {
        return size;
    }
    public void setSize(String size) {
        this.size = size;
    }
    public String getMaterial() {
        return material;
    }
    public void setMaterial(String material) {
        this.material = material;
    }
    public String getGender() {
        return gender;
    }
    public void setGender(String gender) {
        this.gender = gender;
    }
    

}

File: cpp/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"

class Clothing : public Product {
private:
    std::string size;
    std::string material;
    std::string gender;

public:
    // Constructor
    Clothing(const std::string& id = "", const std::string& name = "", const std::string& brand = "",
             double price = 0.0, const std::string& size = "", const std::string& material = "",
             const std::string& gender = "")
        : Product(id, name, brand, price), size(size), material(material), gender(gender) {}

    // Getter methods
    std::string getSize() const { return size; }
    std::string getMaterial() const { return material; }
    std::string getGender() const { return gender; }

    // Setter methods
    void setSize(const std::string& newSize) { size = newSize; }
    void setMaterial(const std::string& newMaterial) { material = newMaterial; }
    void setGender(const std::string& newGender) { gender = newGender; }
};
